app/assets/javascripts/components/revisions/revisions_handler.jsx

```diff
@@ -8,17 +8,28 @@ const RevisionHandler = React.createClass({
 
   propTypes: {
     course_id: React.PropTypes.string,
-    course: React.PropTypes.object
+    course: React.PropTypes.object,
+  },
+
+  getInitialState() {
+    return {
+      limit: 50
+    };
   },
 
   componentWillMount() {
-    return ServerActions.fetch('revisions', this.props.course_id);
+    return ServerActions.fetchCourseRevisions(this.props.course_id, this.state.limit);
   },
 
   sortSelect(e) {
     return UIActions.sort('revisions', e.target.value);
   },
 
+  showMore() {
+    this.setState({ limit: this.state.limit + 100 });
+    return ServerActions.fetchCourseRevisions(this.props.course_id, (this.state.limit + 50));
```

Comment:
I don't understand what is going on here. Why is setState adding 100, but +50 is used in the following line?

Reply:
I agree this solution was rather confusing - tbh I was not that happy with it. I just commited a new one, let me know if you think it's better. :)

Reply:
Huh... I didn't think this would work, but it does... I guess it's possible to assign to this.state directly, even though it's considered "wrong": https://reactjs.org/docs/state-and-lifecycle.html#using-state-correctly

But I don't think we need to treat the first 'more' any differently from the rest. Here's what I suggest:

```
const newLimit = this.state.limit + 100;
this.setState({ limit: newLimit });
return ServerActions.fetchCourseRevisions(this.props.course_id, newLimit);
```